examples/microwave-oven-app/microwave-oven-common/include/operational-state-delegate.h

```diff
@@ -0,0 +1,149 @@
+/*
+ *
+ *    Copyright (c) 2023 Project CHIP Authors
+ *    All rights reserved.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#pragma once
+
+#include <app-common/zap-generated/cluster-objects.h>
+#include <app/clusters/operational-state-server/operational-state-server.h>
+#include <app/util/af-enums.h>
+#include <protocols/interaction_model/StatusCode.h>
+
+namespace chip {
+namespace app {
+namespace Clusters {
+
+class MicrowaveOvenDevice;
```

Comment:
This is a co-dependency between classes, and is not needed. A `void*` could be used, and `MicrowaveOvenDevice * mPauseMicrowaveOvenInstance;` --> `void * mPauseMicrowaveOvenContext;`. With that rename, there would be no need to force the context passed to be an instance, or for this empty pre-declaration to be present.

Reply:
But we stiil need the `MicrowaveOvenDevice` class to define function pointer(e.g. `typedef void (MicrowaveOvenDevice::*HandleOpStateCommand)(Clusters::OperationalState::GenericOperationalError & err);`), the function pointer need to specify namespaces(refer to rvc-app).

Reply:
We have changed to use std::bind & std::function methods to pass function pointer to MicrowaveOvenControl class. In this case, there is no need to pass an instance and pre-declare `class MicrowaveOvenDevice`.